src/query_analytics.cpp

```diff
@@ -57,11 +57,39 @@ void QueryAnalytics::serialize_as_docs(std::string& docs) {
     std::shared_lock lk(lmutex);
 
     std::string key_buffer;
+    nlohmann::json doc;
     for(auto it = local_counts.begin(); it != local_counts.end(); ++it) {
+        std::vector<std::string> results;
+
         it.key(key_buffer);
-        nlohmann::json doc;
+        //sample string : Nike+filter=Air+tags=Colaba,Bandra
+        StringUtils::split(key_buffer, results, "+");
+        if(meta_fields_analytics) {
```

Comment:
This is a bit complex and is necessitated only because the `local_counts` map is using `std::string` as the key. We should instead have a struct as the key of the local_counts map. The struct could look like:

```
struct analytics_params_t {
  std::string query;
  std::map<std::string, std::string> meta;
}
```
We can then have a custom hash function for the map. 

Both `filter_by` and `tag` can go into the meta map. Also, we decided that a single tag string is enough: no need to support array of tags. So let's call it as `analytics_tag`.